### PR TITLE
Compile errors in RendererOMX.cpp

### PR DESCRIPTION
**Pull request by zpon**
I found a few compile-time errors in RendererOMX.cpp (see https://github.com/xbmc/xbmc/commit/94352105d8c68631d90107669d99578f64674431), which I have tried to fix.

I'm not too sure about the changes, and hope somebody can help me review them.
The `bool CRendererOMX::Supports(ESCALINGMETHOD mode)`-method was missing, so I added it with `return false;`, I have no idea whether this is correct.

### CODE REVIEW

File: xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererOMX.cpp [hunk 1]
```diff
@@ -27,6 +27,7 @@
 #include "utils/GLUtils.h"
 #include "settings/MediaSettings.h"
 #include "windowing/WindowingFactory.h"
+#include "DVDCodecs/Video/DVDVideoCodecIMX.h"
```
Comment (anaconda) [inline xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererOMX.cpp]: `DVDVideoCodecOpenMax.h`?

Comment (zpon) [inline xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererOMX.cpp]: Without this I got the following exception:
```
RendererOMX.cpp:64:22: error: invalid use of incomplete type 'class CDVDVideoCodecIMXBuffer'
     picture.IMXBuffer->Lock();
```
From line 67 (patched version).

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererOMX.cpp]: I think this file should be named RenderOpenMax.cpp and it should include VIdeoCodecOpenMax.h

Comment (zpon) [inline xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererOMX.cpp]: @FernetMenta I'm unable to find a file named "VideoCodecOpenMax.h" are you referring to "DVDVideoCodecOpenMax.h"?
I don't have an opinion about naming (other than I think it should be "Renderer"-something not "Render"-something, to match the naming of the other files https://github.com/xbmc/xbmc/tree/master/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender). But should the decision be approved in any way?

Comment (fritsch) [inline xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererOMX.cpp]: OpenMax should not have anything to do with the IMX decoder / render which is for IMX6 only.

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererOMX.cpp]: Yes, currently it is DVDVideoCodecOpenMax and after "DVD" has been dropped VideoCodecOpenMax remains. Sorry for my typo, you are right. It should be RendererOpenMax.

Comment (zpon) [inline xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererOMX.cpp]: If I use `#include "DVDCodecs/Video/DVDVideoCodecOpenMax.h"` instead i get the error below. Could this be related to my hardware? I'm compiling on a cubox-i.

```
$ make
CPP     xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererOMX.o
RendererOMX.cpp: In member function 'virtual void CRendererOMX::AddVideoPictureHW(DVDVideoPicture&, int)':
RendererOMX.cpp:71:22: error: invalid use of incomplete type 'class CDVDVideoCodecIMXBuffer'
     picture.IMXBuffer->Lock();
                      ^
In file included from /home/builder/xbmc-imx-git/src/xbmc/xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGLES.h:35:0,
                 from RendererOMX.h:27,
                 from RendererOMX.cpp:21:
/home/builder/xbmc-imx-git/src/xbmc/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodec.h:55:7: note: forward declaration of 'class CDVDVideoCodecIMXBuffer'
 class CDVDVideoCodecIMXBuffer;
       ^
../../../../../Makefile.include:97: recipe for target 'RendererOMX.o' failed
make: *** [RendererOMX.o] Error 1
```

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererOMX.cpp]: anything named "IMX" in this file is most likely wrong: for reference check: https://github.com/xbmc/xbmc/blob/Jarvis/xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp

Comment (zpon) [inline xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererOMX.cpp]: @FernetMenta Thanks, by that logic, does lines 64-67 even make sense then?

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererOMX.cpp]: I think this is copy/paste error. This has not been tested after the refactoring.

**Comment by zpon** [comment]
I have changed the pull request to accommodate the requested changes. Please let me know if you see any other problems.

**Comment by FernetMenta** [comment]
thanks. does it work again?

**Comment by zpon** [comment]
Well, I'm able to compile the code and launch Kodi. I haven't tried playing anything with it. It will probably require somebody with a working knowledge of OpenMax to tell if it works.

**Comment by FernetMenta** [comment]
anyway, it is an improvement. thanks again
jenkins build and merge

**Pull request merged**